docs/authentication.md

```diff
@@ -33,22 +33,62 @@ sequenceDiagram
+    end
+    loop Refresh backend token every backend access token lifetime
+        AuthService ->> Backend: Refresh token (GET auth/keycloak)
+        Backend ->> AuthService: Response with new backend access token in cookies and new accessExp
     end
-    loop Refresh token every 1/3 of tokens lifetime
+    loop Refresh keycloak token every 1/3 of keycloak access tokens lifetime
         AuthService ->> Keycloak: Refresh token
-        Keycloak ->> AuthService: Updated token
+        Keycloak ->> AuthService: Updated access token and refresh token
     end
     User ->> AuthService: Logout
+    AuthService ->> Keycloak: Logout (GET /auth/realms/{our-realm}/protocol/openid-connect/logout)
     AuthService ->> Backend: Invalidate session (DELETE /auth/logout)
     Backend ->> AuthService: Response with cookies deleted
-    AuthService ->> Keycloak: Logout
     AuthService ->> User: Clear user data
 ```
 
-## Non-Keycloak
+-   **User requests login**: User initiates login with SSO.
+-   **Redirect to Keycloak**: AuthService redirects user to Keycloak login.
+-   **User login with credentials**: User provides credentials on Keycloak.
+-   **Authenticated token returned**: Keycloak provides AuthService with an authenticated token.
+-   **Token access check**: AuthService checks token for access to Yaptide (token includes PLG_YAPTIDE_ACCESS value)
+
+-   **User access verification**:
+
+    -   **If user has access**:
+        AuthService verifies token with Backend.
+        Backend confirms token with Keycloak.
+        -   **If verified**:
+            Keycloak validates the token signature.
+            Backend provides access expiration (`accessExp`) to AuthService.
+            AuthService sets token refresh interval based on `accessExp` (backend token valid time).
+            User receives authentication context.
+        -   **If verification fails**: Backend sends "Forbidden" error due to invalid/expired token or Keycloak error.
+    -   **If user lacks access**: AuthService informs user of access denial.
+
+-   **Backend connection check**: Every 3 minutes, AuthService checks backend availability.
+-   **Backend token periodic refresh**:
+
+    -   Every backend token lifetime (`accessExp`), AuthService requests a token refresh from Backend.
+    -   Backend responds with a new access token in cookies and an updated `accessExp`.
+
+-   **Keycloak token periodic refresh**:
+
+    -   Every 1/3 of the keycloak access token's lifetime, AuthService refreshes the Keycloak token.
+    -   Keycloak provides an updated access token and refresh token to AuthService.
+
+-   **Logout process**:
+    -   User initiates logout.
+    -   AuthService logs user out of Keycloak and clears user data.
+    -   AuthService invalidates Backend session.
+    -   Backend deletes session cookies.
+
+## Our own database of users validation
 
 Overview of login and logout process while in demo or dev modes
```

Comment:
BTW - do we have login process in demo mode ? I though that demo mode means login disables, like in github pages deployment,

Reply:
i meant dev deployment with docker. will fix